platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java

```diff
@@ -253,6 +268,50 @@ public void getPostLogoutRequest() throws Exception {
         response.getEntity().toString().contains(postLogoutUrl));
   }
 
+  @Test
+  public void soapLogoutRequestIssueInstantOld() throws Exception {
+    HttpSessionInvalidator httpSessionInvalidator = mock(HttpSessionInvalidator.class);
+    logoutRequestService.setHttpSessionInvalidator(httpSessionInvalidator);
+
+    InputStream requestStream =
+        LogoutRequestServiceTest.class.getResourceAsStream("/SAMLSoapLogoutRequest-good.xml");
+    Response response = logoutRequestService.soapLogoutRequest(requestStream, null);
+    assertThat(response.getStatus(), is(Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+    verify(httpSessionInvalidator, times(0)).invalidateSession(anyString(), any());
+  }
+
+  @Test
+  public void soapLogoutRequestBadSignature() throws Exception {
+    HttpSessionInvalidator httpSessionInvalidator = mock(HttpSessionInvalidator.class);
+    logoutRequestService.setHttpSessionInvalidator(httpSessionInvalidator);
+
+    LogoutResponse logoutResponse = mock(LogoutResponse.class);
+    doReturn(logoutResponse)
+        .when(logoutMessage)
+        .buildLogoutResponse(anyString(), anyString(), anyString());
+    doThrow(SignatureException.class)
+        .when(simpleSign)
+        .validateSignature(any(Signature.class), any(Document.class));
+
+    InputStream requestStream =
+        LogoutRequestServiceTest.class.getResourceAsStream("/SAMLSoapLogoutRequest-good.xml");
+    Response response = logoutRequestService.soapLogoutRequest(requestStream, null);
+    assertThat(response.getStatus(), is(Status.OK.getStatusCode()));
+    verify(httpSessionInvalidator, times(0)).invalidateSession(anyString(), any());
+  }
+
+  @Test
+  public void soapLogoutRequestNotALogout() throws Exception {
+    HttpSessionInvalidator httpSessionInvalidator = mock(HttpSessionInvalidator.class);
+    logoutRequestService.setHttpSessionInvalidator(httpSessionInvalidator);
+
+    InputStream requestStream =
+        LogoutRequestServiceTest.class.getResourceAsStream("/SAMLSoapLogoutRequest-bad.xml");
+    Response response = logoutRequestService.soapLogoutRequest(requestStream, null);
+    assertThat(response.getStatus(), is(Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+    verify(httpSessionInvalidator, times(0)).invalidateSession(anyString(), any());
```

Comment:
❓ I'm not fully sure what's going on here, but wouldn't we also want a test where we do call `invalidateSession` or are those cases tested elsewhere?

Reply:
itests covering that. Unit testing that would be a matter of testing mocks more than the code itself.